src/utilities/transition/MoveTransition.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+// Required modules.
+var Events = require( '../../mixins/Events.js' );
+var BaseTransition = require( './BaseTransition' );
+var fnBind = require( '../function-bind' ).bind;
+
+// Exported constants.
+var CLASSES = {
+  BASE_CLASS:     'u-move-transition',
+  MOVE_TO_ORIGIN: 'u-move-to-origin',
+  MOVE_LEFT:      'u-move-left',
+  MOVE_LEFT_2X:   'u-move-left-2x',
+  MOVE_LEFT_3X:   'u-move-left-3x',
+  MOVE_RIGHT:     'u-move-right',
+  MOVE_UP:        'u-move-up'
+};
+
+/**
+ * MoveTransition
+ * @class
+ *
+ * @classdesc Initializes new MoveTransition behavior.
+ *
+ * @param {HTMLNode} element
+ *   DOM element to apply move transition to.
+ * @returns {MoveTransition} An instance.
+ */
+function MoveTransition( element ) { // eslint-disable-line max-statements, no-inline-comments, max-len
+
+  var _baseTransition = new BaseTransition( element, CLASSES );
+
+  /**
+   * @returns {MoveTransition} An instance.
+   */
+  function init() {
+    _baseTransition.init();
+    var _transitionCompleteBinded = fnBind( _transitionComplete, this );
+    _baseTransition.addEventListener( BaseTransition.END_EVENT,
+                                      _transitionCompleteBinded );
+    return this;
+  }
+
+  /**
+   * Handle the end of a transition.
+   */
+  function _transitionComplete() {
+    this.trigger( BaseTransition.END_EVENT, { target: this } );
+  }
+
+  /**
+   * Move to the element's original coordinates.
+   * @returns {MoveTransition} An instance.
+   */
+  function moveToOrigin() {
+    _baseTransition.applyClass( CLASSES.MOVE_TO_ORIGIN );
+
+    return this;
+  }
+
+  /**
+   * Move to the left by applying a utility move class.
+   * @param {Number} count
+   *   How many times to move left as a multiplication of the element's width.
+   * @returns {MoveTransition} An instance.
```

Comment:
Forgot to include `@throws {Error} When supplied count value is out of range of available CSS classes.`